jsonschema/validators.py

```diff
@@ -259,17 +267,40 @@ def from_schema(cls, schema, *args, **kwargs):
 
         return cls(schema.get(u"id", u""), schema, *args, **kwargs)
 
+    def push_scope(self, scope):
+        self.scopes_stack.append(
+            self.urljoin_cache(self.resolution_scope, scope))
+
+    def pop_scope(self):
```

Comment:
Same comment about making these methods private, unless there's a reason you can think of to not (although I thikn this question is moot if we consider the composition route I mentioned above) -- certainly if it's not private we probably should handle errors a bit nicer, which might make sense regardless.